circle.yml

```diff
@@ -0,0 +1,32 @@
+machine:
+  environment:
+    GRADLE_OPTS: '-Dorg.gradle.jvmargs="-Xmx2048m -XX:+HeapDumpOnOutOfMemoryError"'
+
+dependencies:
+  pre:
+    - mkdir $ANDROID_HOME/licenses
+    - cp -r licenses/. $ANDROID_HOME/licenses
+    - echo y | android update sdk --no-ui -a --filter tools
+    - echo y | android update sdk --no-ui -a --filter platform-tools
+    - echo y | android update sdk --no-ui -a --filter build-tools-26.0.1
+    - echo y | android update sdk --no-ui -a --filter extra-google-google_play_services
+    - echo y | android update sdk --no-ui -a --filter extra-google-m2repository
+    - echo y | android update sdk --no-ui -a --filter extra-android-m2repository
+
+test:
+  override:
+    - ./gradlew testAllVersionsProductionDebugUnitTest
+    - ./gradlew testAllVersionsProductionReleaseUnitTest
+    - ./gradlew testAllVersionsStagingReleaseUnitTest
+    - ./gradlew testAllVersionsStagingDebugUnitTest
+    - ./gradlew lintAllVersionsProductionRelease
+    - ./gradlew lintAllVersionsProductionDebug
+    - ./gradlew lintAllVersionsStagingDebug
+    - ./gradlew lintAllVersionsStagingRelease
+  post:
+    - mkdir -p $CIRCLE_TEST_REPORTS/junit/
+    - find . -type f -regex ".*/build/test-results/.*xml" -exec cp {} $CIRCLE_TEST_REPORTS/junit/ \;
+    - mkdir -p $CIRCLE_TEST_REPORTS/detekt/
+    - find . -type f -regex ".*/templete/app/build/reports/detekt/.*xml" -exec cp {} $CIRCLE_TEST_REPORTS/detekt/ \;
+    - mkdir -p $CIRCLE_TEST_REPORTS/lints/
```

Comment:
And the directories creation too :smile: : `mkdir -p $CIRCLE_TEST_REPORTS/{junit,detekt,lints}`